File: Examples/VisualBasic/AsposePdfFacades/Images/ConvertPageRegion.vb
Imports System.IO
Imports System
Imports Aspose.Pdf
Imports Aspose.Pdf.Facades

Namespace AsposePDFFacades.Images
    Public Class ConvertPageRegion
        Public Shared Sub Run()
            ' ExStart:ConvertPageRegion
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdfFacades_Images()
            ' Instantiate PdfPageEditor class to get particular page region
            Dim editor As New Aspose.Pdf.Facades.PdfPageEditor()
            ' Bind the source PDF file
            editor.BindPdf(dataDir & Convert.ToString("Convert-PageRegion.pdf"))
            ' Move the origin of PDF file to particular point
            editor.MovePosition(0, 700)
            ' Create a memory stream object
            Dim ms As New MemoryStream()
            ' Save the updated document to stream object
            editor.Save(ms)
            ' Create PdfConverter object
            Dim objConverter As New PdfConverter()
            ' Bind input pdf file
            objConverter.BindPdf(ms)
            ' Set StartPage and EndPage properties to the page number to
            ' You want to convert images from
            objConverter.StartPage = 1
            objConverter.EndPage = 1
            ' Counter
            Dim page As Integer = 1
            ' Initialize the converting process
            objConverter.DoConvert()
            ' Check if pages exist and then convert to image one by one
            While objConverter.HasNextImage()
                objConverter.GetNextImage((dataDir & Convert.ToString("Specific_Region-Image")) + System.Math.Max(System.Threading.Interlocked.Increment(page), page - 1).ToString() + "_out_.jpeg", System.Drawing.Imaging.ImageFormat.Jpeg)
            End While
            ' Close the PdfConverter object
            objConverter.Close()
            ' Close MemoryStream object holding the updated document
            ms.Close()
            ' ExEnd:ConvertPageRegion

        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePdfFacades/Images/ConvertPageRegion.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Facades;

namespace Aspose.Pdf.Examples.CSharp.AsposePDFFacades.Images
{
    public class ConvertPageRegion
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdfFacades_Images();
            // instantiate PdfPageEditor class to get particular page region
            Aspose.Pdf.Facades.PdfPageEditor editor = new Aspose.Pdf.Facades.PdfPageEditor();
            // bind the source PDF file
            editor.BindPdf(dataDir + "SampleInput.pdf");
            // move the origin of PDF file to particular point
            editor.MovePosition(0, 700);
            // create a memory stream object
            MemoryStream ms = new MemoryStream();
            // save the updated document to stream object
            editor.Save(ms);
            //create PdfConverter object
            PdfConverter objConverter = new PdfConverter();
            //bind input pdf file
            objConverter.BindPdf(ms);
            //set StartPage and EndPage properties to the page number to
            //you want to convert images from
            objConverter.StartPage = 1;
            objConverter.EndPage = 1;
            //Counter
            int page = 1;
            //initialize the converting process
            objConverter.DoConvert();
            //check if pages exist and then convert to image one by one
            while (objConverter.HasNextImage())
                objConverter.GetNextImage(dataDir+ "Specific_Region-Image" + page++ + ".jpeg", System.Drawing.Imaging.ImageFormat.Jpeg);
            //close the PdfConverter object
            objConverter.Close();
            // close MemoryStream object holding the updated document
            ms.Close();
            
            
        }
    }
}